Stack/StackTest.h

```diff
@@ -0,0 +1,30 @@
+#pragma once
+
+template <typename>
+class IStack;
```

Comment:
Why do you declare a forward declaration of the interface? We can include the file here. 
Is because the performance to avoid including all the interface declarations here?

Reply:
Not because performance. Is a good practice to keep the include tree clean. Headers should never include anything.

Reply:
Got it. I understand why. Thanks.